src/allmydata/web/status.py

```diff
@@ -673,90 +673,129 @@ def render_progress(self, ctx, data):
 
-    def render_started(self, ctx, data):
-        started_s = render_time(data.get_started())
-        return started_s
+# Renders "/status/retrieve-%d".
+class RetrieveStatusPage(MultiFormatResource):
 
-    def render_si(self, ctx, data):
-        si_s = base32.b2a_or_none(data.get_storage_index())
+    def __init__(self, retrieve_status):
+        """
+        :retrieve_status retrieve.RetrieveStatus: stats provider.
+        """
+        super(RetrieveStatusPage, self).__init__()
+        self._retrieve_status = retrieve_status
+
+    def render_HTML(self, req):
+        elem = RetrieveStatusElement(self._retrieve_status)
+        return renderElement(req, elem)
+
+
+class RetrieveStatusElement(Element):
+
+    loader = XMLFile(FilePath(__file__).sibling("retrieve-status.xhtml"))
+
+    def __init__(self, retrieve_status):
+        super(RetrieveStatusElement, self).__init__()
+        self._retrieve_status = retrieve_status
+
+    @renderer
+    def started(self, req, tag):
+        started_s = render_time(self._retrieve_status.get_started())
+        return tag(started_s)
+
+    @renderer
+    def si(self, req, tag):
+        si_s = base32.b2a_or_none(self._retrieve_status.get_storage_index())
         if si_s is None:
             si_s = "(None)"
-        return si_s
+        return tag(si_s)
 
-    def render_helper(self, ctx, data):
-        return {True: "Yes",
-                False: "No"}[data.using_helper()]
+    @renderer
+    def helper(self, req, tag):
+        return tag({True: "Yes",
+                    False: "No"}[self._retrieve_status.using_helper()])
 
-    def render_current_size(self, ctx, data):
-        size = data.get_size()
+    @renderer
+    def current_size(self, req, tag):
+        size = str(self._retrieve_status.get_size())
         if size is None:
             size = "(unknown)"
-        return size
+        return tag(size)
 
-    def render_progress(self, ctx, data):
-        progress = data.get_progress()
+    @renderer
+    def progress(self, req, tag):
+        progress = self._retrieve_status.get_progress()
         # TODO: make an ascii-art bar
-        return "%.1f%%" % (100.0 * progress)
+        return tag("%.1f%%" % (100.0 * progress))
 
-    def render_status(self, ctx, data):
-        return data.get_status()
+    @renderer
+    def status(self, req, tag):
+        return tag(self._retrieve_status.get_status())
 
-    def render_encoding(self, ctx, data):
-        k, n = data.get_encoding()
-        return ctx.tag["Encoding: %s of %s" % (k, n)]
+    @renderer
+    def encoding(self, req, tag):
+        k, n = self._retrieve_status.get_encoding()
+        return tag("Encoding: %s of %s" % (k, n))
 
-    def render_problems(self, ctx, data):
-        problems = data.get_problems()
+    @renderer
+    def problems(self, req, tag):
+        problems = self._retrieve_status.get_problems()
         if not problems:
             return ""
-        l = T.ul()
+        l = tags.ul()
```

Comment:
maybe at least `ul`? `l` can be confused with `1` easily for many typefaces

Reply:
Renamed `l` to `ul` in 5aa0dd4.